app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -281,7 +281,11 @@ def orchestration_stacks
       o.status_reason = stack.stack_status_reason
       o.parent = persister.orchestration_stacks.lazy_find(stack.parent)
       o.orchestration_template = orchestration_template(stack)
-      o.cloud_tenant = persister.cloud_tenants.lazy_find(stack.service.current_tenant["id"])
```

Comment:
I think we should keep the original tenant_id lookup.

URI parsing looks quite strange, it would be better write it more robust and use just in case stack.service.current_tenant["id"] doesn't work.

Reply:
Agreed, but I'd prefer to see `OS::project_id` checked first, and then if that's unavailable fall back to something else. `stack.service.current_tenant["id"]` is afaik always going to be the tenant used to connect to Heat, so that will usually not be correct.

Reply:
@alexander-demichev does the url in `links` actually have the stack's project_id, or is it just the project_id of the user who made the API request?

Reply:
@mansam I need to check how url is generated, but can we handle empty parameters other way?

Reply:
@alexander-demichev I think the ultimate fallback if the project id can't be found would be to just use `stack.service.current_tenant["id"]` so that it's assigned to the admin.